core/arch/arm/plat-sam/drivers/clk/at91_pll.c

```diff
@@ -0,0 +1,303 @@
+// SPDX-License-Identifier: GPL-2.0-or-later or BSD-3-Clause
+/*
+ *  Copyright (C) 2013 Boris BREZILLON <b.brezillon@overkiz.com>
+ *  Copyright (C) 2021 Microchip
+ */
+
+#include <io.h>
+#include <kernel/delay.h>
+#include <kernel/panic.h>
+#include <util.h>
+#include <mm/core_memprot.h>
+#include <types_ext.h>
+
+#include "at91_clk.h"
+
+#define PLL_STATUS_MASK(id)	(1 << (1 + (id)))
+#define PLL_REG(id)		(AT91_CKGR_PLLAR + ((id) * 4))
+#define PLL_DIV_MASK		0xff
+#define PLL_DIV_MAX		PLL_DIV_MASK
+#define PLL_DIV(reg)		((reg) & PLL_DIV_MASK)
+#define PLL_MUL(reg, layout)	(((reg) >> (layout)->mul_shift) & \
+				 (layout)->mul_mask)
+#define PLL_MUL_MIN		2
+#define PLL_MUL_MASK(layout)	((layout)->mul_mask)
+#define PLL_MUL_MAX(layout)	(PLL_MUL_MASK(layout) + 1)
+#define PLL_ICPR_SHIFT(id)	((id) * 16)
+#define PLL_ICPR_MASK(id)	(0xffff << PLL_ICPR_SHIFT(id))
+#define PLL_MAX_COUNT		0x3f
+#define PLL_COUNT_SHIFT		8
+#define PLL_OUT_SHIFT		14
+#define PLL_MAX_ID		1
+
+struct clk_pll {
+	vaddr_t base;
+	uint8_t id;
+	uint8_t div;
+	uint8_t range;
+	uint16_t mul;
+	const struct clk_pll_layout *layout;
+	const struct clk_pll_characteristics *characteristics;
+};
+
+static inline bool clk_pll_ready(vaddr_t base, int id)
+{
+	unsigned int status = io_read32(base + AT91_PMC_SR);
+
+	return status & PLL_STATUS_MASK(id) ? 1 : 0;
```

Comment:
`return io_read32(base + AT91_PMC_SR) & PLL_STATUS_MASK(id);` does the job.

Reply:
I guess this was a precaution for the bool cast also.